scenarios/erc20_multichain_factory/test/DynamicContractRecovery_test.res

```diff
@@ -274,7 +274,7 @@ describe("Dynamic contract restart resistance test", () => {
       )
 
       let restartedFetchState =
-        restartedChainFetcher.fetchState.partitions->List.head->Option.getExn
+        restartedChainFetcher.fetchState.partitions->Js.Dict.values->Array.get(0)->Option.getExn
```

Comment:
Since we have numerical ids, I think it's better to turn partitions into an array? This way we still be able to get items by id - aka hashmap + no need to always call Js.Dict.values

Reply:
The `newestPartitionIndex` can be calculated from length

Reply:
We could, the operation that I want to improve is updating a single partition in the array in an "immutable" way.

So I guess we can shallow copy the array and then set the given index on the array.

Reply:
@DZakh, I just made some benchmarks here: https://jsbm.dev/Gwg2BVoQDxavT

Super interesting. Object.assign is the fastest of the 4 cases.

Reply:
No, it's the slowest :)

Reply:
Oh lol 🤣 I was reading the bars as runtime not ops/s

Reply:
Sweet, I like the idea of using arrays 👍🏼